gumby/modules/dispersy_module.py

```diff
@@ -0,0 +1,166 @@
+from os import environ, path, getpid
+from collections import defaultdict
+from time import time
+
+from twisted.python.log import addObserver
+from twisted.internet import reactor
+from twisted.internet.defer import inlineCallbacks
+from twisted.internet.task import deferLater
+
+from gumby.sync import experiment_callback
+from gumby.modules.experiment_module import static_module
+from gumby.modules.base_dispersy_module import BaseDispersyModule
+from gumby.modules.isolated_community_loader import IsolatedCommunityLoader
+from gumby.modules.gumby_session import GumbySession
+
+from Tribler.Core.SessionConfig import SessionStartupConfig
+from Tribler.dispersy.dispersy import Dispersy
+from Tribler.dispersy.crypto import ECCrypto, NoCrypto
+from Tribler.dispersy.endpoint import StandaloneEndpoint
+from Tribler.dispersy.util import unhandled_error_observer
+from Tribler.dispersy.candidate import CANDIDATE_STUMBLE_LIFETIME, CANDIDATE_WALK_LIFETIME, CANDIDATE_INTRO_LIFETIME
+
+
+@static_module
+class DispersyModule(BaseDispersyModule):
+    def __init__(self, experiment):
+        super(DispersyModule, self).__init__(experiment)
+        self._crypto = self._initialize_crypto()
+        self.session_id = environ['SYNC_HOST'] + environ['SYNC_PORT']
+        self.custom_community_loader = self.create_community_loader()
+
+    def on_id_received(self):
+        super(DispersyModule, self).on_id_received()
+        self.session_config = self.setup_session_config()
+        self.session = GumbySession(scfg=self.session_config)
+
+    @experiment_callback
+    def start_session(self):
+        self._logger.debug("Starting dispersy")
+        self.dispersy = Dispersy(StandaloneEndpoint(self.session_config.get_dispersy_port(), '0.0.0.0'),
+                                  u'.', u"dispersy.db", self._crypto)
+        self.dispersy.statistics.enable_debug_statistics(True)
+        self.dispersy.start(autoload_discovery=False)
+
+        self.custom_community_loader.load(self.dispersy, self.session)
+
+        self._do_log()
+        self._logger.debug("Finished starting dispersy")
+
+    @experiment_callback
+    def stop_session(self):
+        self.dispersy.stop()
+
+    @experiment_callback
+    def churn(self, *args):
+        self.print_on_change('community-churn', {}, {'args':args})
+
+    def _initialize_crypto(self):
+        if environ.get('TRACKER_CRYPTO', 'ECCrypto') == 'ECCrypto':
```

Comment:
After grepping, this turned out the be the only use for TRACKER_CRYPTO, appart from the obvious setting of the crypto of the tracker. I think this is mostly to make the dispersy trackers compatible with the dispersy instances in the experiment clients. So considering this, it's not the worst problem in the world. I will change this to an experiment callback that can be used to configure what crypto to use when starting dispersy.